core/src/test/java/nl/nn/adapterframework/pipes/ForEachChildElementPipeTest.java

```diff
@@ -897,7 +901,48 @@ public void testNamespacedXPath() throws Exception {
 		assertEquals(expected, actual);
 	}
 
-	
+	//The XmlFileElementIteratorPipe has been deprecated, elementName has been replaced by targetElement.
+	//This test proves that the old method works with the ForEachChildElemenPipe
+	@Test
+	public void testElementName() throws Exception {
+		XsltSender sender = new XsltSender();
+		sender.setXpathExpression("concat(Person/PersonName/Id,'_',Person/Demographics/Gender)");
+		pipe.setProcessFile(true);
+		pipe.setSender(sender);
+		pipe.setTargetElement("Person");
+		pipe.configure();
+		pipe.start();
+		
+		URL input = TestFileUtils.getTestFileURL("/XmlFileElementIteratorPipe/input.xml");
+		File file = new File(input.toURI());
+		String expected = TestFileUtils.getTestFile("/XmlFileElementIteratorPipe/ElementNameOutput.xml");
+		PipeRunResult prr = doPipe(pipe, file.toString(), session);
+		String result = Message.asString(prr.getResult());
+		TestAssertions.assertEqualsIgnoreCRLF(expected, result);
+	}
+
+	// The XmlFileElementIteratorPipe has been deprecated, elementChain has been replaced with targetElement + containerElement.
+	// This test proves that the old method works with the ForEachChildElemenPipe.
+	// The xPath in the sender was adjusted to match the targetElement
+	@Test
+	public void testElementChain() throws Exception {
+		XsltSender sender = new XsltSender();
+		sender.setXpathExpression("concat(Party/Person/PersonName/Id,'_',Party/Person/Demographics/Gender)");
+		pipe.setProcessFile(true);
+		pipe.setSender(sender);
+		pipe.setContainerElement("PartyInternalAgreementRole");
```

Comment:
In deze setup heb je de setting voor containerElement niet nodig, want er zijn geen parties buiten PartyInternalAgreementRole